diff --git a/hw/display/virtio-gpu-virgl.c b/hw/display/virtio-gpu-virgl.c
index 6e298f997d..62ede84a77 100644
--- a/hw/display/virtio-gpu-virgl.c
+++ b/hw/display/virtio-gpu-virgl.c
@@ -15,6 +15,7 @@
 #include "qemu/error-report.h"
 #include "qemu/iov.h"
 #include "trace.h"
+#include "qemu/drm.h"
 #include "hw/virtio/virtio.h"
 #include "hw/virtio/virtio-gpu.h"
 #include "hw/virtio/virtio-gpu-bswap.h"
@@ -1341,6 +1342,15 @@ static int virgl_make_context_current(void *opaque, int scanout_idx,
     return qemu_console_gl_ctx_make_current(g->parent_obj.scanout[scanout_idx].con, qctx);
 }
 
+static int virgl_get_drm_fd(void *opaque)
+{
+    int fd = -1;
+
+    fd = qemu_drm_rendernode_open(NULL);
+
+    return fd;
+}
+
 static struct virgl_renderer_callbacks virtio_gpu_3d_cbs = {
 #if VIRGL_VERSION_MAJOR >= 1
     .version             = 3,
@@ -1354,6 +1364,7 @@ static struct virgl_renderer_callbacks virtio_gpu_3d_cbs = {
 #if VIRGL_VERSION_MAJOR >= 1
     .write_context_fence = virgl_write_context_fence,
 #endif
+    .get_drm_fd          = virgl_get_drm_fd,
 };
 
 static void virtio_gpu_print_stats(void *opaque)
@@ -1434,7 +1445,7 @@ static bool virtio_gpu_virgl_reset(VirtIOGPU *g)
 static int virtio_gpu_virgl_init(VirtIOGPU *g)
 {
     int ret;
-    uint32_t flags = 0;
+    uint32_t flags = VIRGL_RENDERER_USE_VIDEO;
     VirtIOGPUGL *gl = VIRTIO_GPU_GL(g);
 
 #if VIRGL_RENDERER_CALLBACKS_VERSION >= 4
diff --git a/meson.build b/meson.build
index 40e3f4ec44..2cde733abc 100644
--- a/meson.build
+++ b/meson.build
@@ -1422,6 +1422,11 @@ if not get_option('virglrenderer').auto() or have_system or have_vhost_user_gpu
                      method: 'pkg-config',
                      required: get_option('virglrenderer'))
 endif
+if virgl.version().version_compare('>= 1.0.0')
+  message('Enabling virglrenderer unstable APIs')
+  virgl = declare_dependency(compile_args: '-DVIRGL_RENDERER_UNSTABLE_APIS -DVIRGL_RENDERER_VENUS',
+                            dependencies: virgl)
+endif
 rutabaga = not_found
 if not get_option('rutabaga_gfx').auto() or have_system or have_vhost_user_gpu
   rutabaga = dependency('rutabaga_gfx_ffi',